demos/python_demos/monitors_extension/monitors_extension.cpp

```diff
@@ -0,0 +1,228 @@
+        {"reportMeans", reinterpret_cast<PyCFunction>(presenter_reportMeans), METH_NOARGS},
+        {"handleKey", reinterpret_cast<PyCFunction>(presenter_handleKey), METH_VARARGS | METH_KEYWORDS},
+        {"drawGraphs", reinterpret_cast<PyCFunction>(presenter_drawGraphs), METH_VARARGS | METH_KEYWORDS},
+        {nullptr}};  // Sentinel
+
+PyObject *presenter_getYPos(PresenterObject *self, void *closure) {
+    if (!self->_presenter) {
+        PyErr_SetString(PyExc_AttributeError, "Underlying C++ presenter is nullptr");
+        return nullptr;
+    }
+    PyObject * yPos = PyLong_FromLong(self->_presenter->yPos);
+    if (PyErr_Occurred())
+        return nullptr;
+    return yPos;
+}
+
+int presenter_setYPos(PresenterObject *self, PyObject *value, void *closure) {
+    PyErr_Format(PyExc_RuntimeError, "Read-only attribute: yPos");
+    return -1;
+}
+
+PyObject *presenter_getGraphSize(PresenterObject *self, void *closure) {
+        if (!self->_presenter) {
+            PyErr_SetString(PyExc_AttributeError, "Underlying C++ presenter is nullptr");
+            return nullptr;
+        }
+        PyObject * width, * height;
+        width = PyLong_FromLong(self->_presenter->graphSize.width);
+        if (PyErr_Occurred())
+            return nullptr;
+        height = PyLong_FromLong(self->_presenter->graphSize.height);
+        if (PyErr_Occurred())
+            return nullptr;
+        return PyTuple_Pack(2, width, height);
+}
+
+int presenter_setGraphSize(PresenterObject *self, PyObject *value, void *closure) {
+    PyErr_Format(PyExc_RuntimeError, "Read-only attribute: graphSize");
+    return -1;
+}
+
+PyObject *presenter_getGraphPadding(PresenterObject *self, void *closure) {
+    if (!self->_presenter) {
+        PyErr_SetString(PyExc_AttributeError, "Underlying C++ presenter is nullptr");
+        return nullptr;
+    }
+    PyObject * graphPadding = PyLong_FromLong(self->_presenter->graphPadding);
+    if (PyErr_Occurred())
+        return nullptr;
+    return graphPadding;
+}
+
+int presenter_setGraphPadding(PresenterObject *self, PyObject *value, void *closure) {
+    PyErr_Format(PyExc_RuntimeError, "Read-only attribute: graphPadding");
+    return -1;
+}
+
+char graphPaddingName[] = "graphPadding";
+PyGetSetDef presenter_getsetters[] = {
+    {yPosName, (getter)presenter_getYPos, (setter)presenter_setYPos,},
+    {graphSizeName, (getter)presenter_getGraphSize, (setter)presenter_setGraphSize},
+    {graphPaddingName, (getter)presenter_getGraphPadding, (setter)presenter_setGraphPadding},
+    {nullptr}};  // Sentinel
+
+char monitors_extension_doc[] = "The module is a wrapper over C++ monitors. It guarantees that C++ and Python "
+    "monitors are consistent.";
+
+PyTypeObject presenterType{PyVarObject_HEAD_INIT(nullptr, 0)
+    "monitors_extension.Presenter",  // tp_name
+    sizeof(PresenterObject),  // tp_basicsize
+    0,  // tp_itemsize
+    reinterpret_cast<destructor>(presenter_dealloc),  // tp_dealloc
```

Comment:
I can't fix similar problem with `PyMethodDef{"handleKey", reinterpret_cast<PyCFunction>(presenter_handleKey), METH_VARARGS | METH_KEYWORDS}` because of different number of arguments. So I wouldn't fix that at all. Moreover [tutorial suggest to cast the func](https://docs.python.org/3/extending/newtypes_tutorial.html) and [CPython does the same](https://github.com/python/cpython/blob/08050e959e6c40839cd2c9e5f6a4fd1513e3d605/Objects/listobject.c#L3028).

Reply:
> Moreover tutorial suggest to cast the func and CPython does the same.

Okay, I can't argue with that (although I still don't like it 😄).

FWIW, while you do have to use a cast with `METH_KEYWORDS`, you could still make the function itself the correct type, so that there would be no UB when Python calls it.

Reply:
This wouldn't be a complete fix.

This different signature with `METH_KEYWORDS` makes me think that CPython never calls the functions exactly how I proved them. It probably casts these pointer to signatures it thinks are suitable (based on other info like flags that I provide), so it doesn't matter what was the original pointer type. So there is even a chance (I didn't check) that internally CPython will cast `presenter_dealloc` to `void (PresenterObject *self)` even if I pass a pointer to `void presenter_dealloc(PyObject*self)`.

Reply:
> So there is even a chance (I didn't check) that internally CPython will cast presenter_dealloc to void (PresenterObject *self)

That's impossible, because CPython doesn't know what `PresenterObject` is.